server_fn/src/codec/json.rs

```diff
@@ -1,221 +1,36 @@
-        })?;
-        Request::try_new_post(path, accepts, Json::CONTENT_TYPE, data)
-    }
-}
-
-impl<E, T, Request> FromReq<Json, Request, E> for T
-where
-    Request: Req<E> + Send + 'static,
-    T: DeserializeOwned,
-    E: FromServerFnError,
-{
-    async fn from_req(req: Request) -> Result<Self, E> {
-        let string_data = req.try_into_string().await?;
-        serde_json::from_str::<Self>(&string_data)
-            .map_err(|e| ServerFnErrorErr::Args(e.to_string()).into_app_error())
-    }
-}
 
-impl<E, T, Response> IntoRes<Json, Response, E> for T
-where
-    Response: TryRes<E>,
-    T: Serialize + Send,
-    E: FromServerFnError,
-{
-    async fn into_res(self) -> Result<Response, E> {
-        let data = serde_json::to_string(&self).map_err(|e| {
-            ServerFnErrorErr::Serialization(e.to_string()).into_app_error()
-        })?;
-        Response::try_from_string(Json::CONTENT_TYPE, data)
-    }
-}
+/// Serializes and deserializes JSON with [`serde_json`].
+pub struct JsonEncoding;
 
-impl<E, T, Response> FromRes<Json, Response, E> for T
-where
-    Response: ClientRes<E> + Send,
-    T: DeserializeOwned + Send,
-    E: FromServerFnError,
-{
-    async fn from_res(res: Response) -> Result<Self, E> {
-        let data = res.try_into_string().await?;
-        serde_json::from_str(&data).map_err(|e| {
-            ServerFnErrorErr::Deserialization(e.to_string()).into_app_error()
-        })
-    }
-}
-
-/// An encoding that represents a stream of JSON data.
-///
-/// A server function that uses this as its output encoding should return [`StreamingJson`]
-///
-/// ## Browser Support for Streaming Input
-///
-/// Browser fetch requests do not currently support full request duplexing, which
-/// means that that they do begin handling responses until the full request has been sent.
-/// This means that if you use a streaming input encoding, the input stream needs to
-/// end before the output will begin.
-///
-/// Streaming requests are only allowed over HTTP2 or HTTP3.
-pub struct StreamingJson;
-
-impl Encoding for StreamingJson {
-    // Each chunk is encoded as a JSON object, but the overall stream is not valid JSON so this uses the default stream content type
-    const CONTENT_TYPE: &'static str = Streaming::CONTENT_TYPE;
-    const METHOD: Method = Streaming::METHOD;
-}
-
-/// A stream of typed data encoded as JSON.
-///
-/// A server function can return this type if its output encoding is [`StreamingJson`].
-///
-/// ## Browser Support for Streaming Input
-///
-/// Browser fetch requests do not currently support full request duplexing, which
-/// means that that they do begin handling responses until the full request has been sent.
-/// This means that if you use a streaming input encoding, the input stream needs to
-/// end before the output will begin.
-///
-/// Streaming requests are only allowed over HTTP2 or HTTP3.
-pub struct JsonStream<T, E>(Pin<Box<dyn Stream<Item = Result<T, E>> + Send>>);
```

Comment:
Note json streaming http support is removed entirely. The browser doesn't respect the http chunk boundary which makes decoding this unreliable: https://github.com/DioxusLabs/dioxus/issues/3695

Reply:
Yeah, unfortunate but understood. The websockets will be a good replacement for that in any case, I think.